python/lsst/pipe/base/task.py

```diff
@@ -122,9 +122,10 @@ def __init__(self, config=None, name=None, parentTask=None, log=None):
             if config is None:
                 config = self.ConfigClass()
             self._taskDict = dict()
-            if log is None:
-                log = pexLog.getDefaultLog()
-            self.log = pexLog.Log(log, self._fullName)
+            if log is not None and isinstance(log, Log):
```

Comment:
Why is the `isinstance` test? This was not present before and I'm very uncomfortable having it now. Is this a way of trying to avoid pex_logging? If so, is the problem serious enough to need avoiding? If so, please warn or raise if the wrong type of log is passed in. I am uncomfortable with code that silently ignores issues, as it leads to mysterious and silent failures.